standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -1577,24 +1590,20 @@ public long findMinTxnIdSeenOpen() throws MetaException {
             }
             minOpenTxn = rs.getLong(1);
             if (rs.wasNull()) {
-              minOpenTxn = -1L;
+              minOpenTxn = Long.MAX_VALUE;
```

Comment:
Can we use ternary operator here - 
`minOpenTxn = rs.wasNull() ? Long.MAX_VALUE : rs.getLong(1)`

Reply:
nope, per javaDoc wasNull reports whether the last column read had a value of SQL <code>NULL</code>. You must first call one of the getter methods on a column to try to read its value and then call the method <code>wasNull</code> to see if the value read was SQL <code>NULL</code>.

Reply:
Ok